ci/check-code-and-docs-gen.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -ex
+
+if [ ! -f .gitignore ]; then
+  echo "_output" > .gitignore
+fi
+
+make install-go-tools
```

Comment:
this runs `go mod download` after `go mod tidy` which may be dangerous in our codegen check https://github.com/golang/go/issues/35832#issuecomment-571799739 perhaps related to the "caching" we saw

i would change the `generated-code` target to have `install-go-tools` as a pre-req and then remove the call to `make install-go-tools`

Reply:
or since we call `go mod tidy` anyways, we don't need it as a pre-req, it already is. we can just remove this